### PR TITLE
384 Use SQLite3 installed package in the build.

### LINKED ISSUE #384: use sqlite3 package
**Describe:.**
Currently `areg-sdk` is using `sqlite3` source codes and compiles as a static library. The `vcpkg` has reference to the `sqlite3` package, which can be as well used in the `areg-sdk`.

**Actions:**
1. make appropriate changes to have possibility to use `sqlite3` package available at `vcpkg`.
2. Keep the existing source codes to use `sqlite3` either when cannot find the package or when developer forces.

### PR DESCRIPTION
**Pull request by aregtech**
Made changes in the cmake files to use existing SQLite3 package instead of compiling static library in the project. Up from now it is possible to use either existing in the system sqlite3 library or compile the existing in the project source.

### CODE REVIEW

File: conf/cmake/install.cmake [hunk 1]
```diff
@@ -65,24 +65,27 @@ if (AREG_INSTALL)
     # Check and install thirdparty dependencies if needed
     if (AREG_INSTALL_DEPENDS)
 
-        # copy compiled binaries in the bin and lib directories
-        install( TARGETS sqlite3
-            ARCHIVE DESTINATION lib         COMPONENT Development   COMPONENT ${AREG_PACKAGE_NAME}
-                    PERMISSIONS OWNER_READ OWNER_WRITE               GROUP_READ WORLD_READ
-        )
+        # Check how SQLite was built
+        if (NOT AREG_FOUND_SQLITE3_PACKAGE)
```
Comment (vukihub) [inline conf/cmake/install.cmake:69]: Maybe makes sence to copy dependencies?

File: conf/cmake/user.cmake [hunk 2]
```diff
@@ -201,6 +203,10 @@ if (NOT DEFINED AREG_INSTALL_DEPENDS)
     option(AREG_INSTALL_DEPENDS "Disable installation of thirdparty dependencies" OFF)
 endif()
 
+if (NOT DEFINED AREG_SQLITE_PACKAGE)
```
Comment (vukihub) [inline conf/cmake/user.cmake:206]: What about integrating "use packages" flag to use GTest package as well and all other upcoming?

File: thirdparty/sqlite3/Readme.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
 # SQLite sources
 
 These are _amalgamation_ sources of the SQLite database, which are taken from the [_Download_](https://sqlite.org/download.html) page of the [SQLite](https://sqlite.org/) site and not owned by Aregtech. For more information about the SQLite and licenses, please visit the official web site.
+
+## Using SQLite package
+
```
Comment (vukihub) [inline thirdparty/sqlite3/Readme.md:35]: would be nice to add more information about compiling with packages.

**Review by vukihub — CHANGES_REQUESTED** [review]
Please see comments.

File: conf/cmake/install.cmake (hunk 1 above)
Comment (aregtech) [reply conf/cmake/install.cmake:69]: Can create a ticket to think about. But not with these changes.

File: conf/cmake/user.cmake (hunk 2 above)
Comment (aregtech) [reply conf/cmake/user.cmake:206]: Good idea. I was thinking about. I'll create a separate ticket for this.

File: thirdparty/sqlite3/Readme.md (hunk 3 above)
Comment (aregtech) [reply thirdparty/sqlite3/Readme.md:35]: OK. Agree.

**Review by vukihub — APPROVED** [review]
Looks fine.

**Pull request merged**